src/middlewared/middlewared/etc_files/truesearch/config.json.py

```diff
@@ -0,0 +1,83 @@
+import json
+import os
+
+from middlewared.plugins.etc import FileShouldNotExist
+from middlewared.utils.smb import TRUESEARCH_ES_PATH
+
+
+def render(service, middleware):
+    if not os.path.exists("/var/db/system/truesearch"):
+        raise FileShouldNotExist()
+
+    directories = middleware.call_sync("truesearch.directories")
+    if not directories:
+        raise FileShouldNotExist()
+
+    os.makedirs("/etc/truesearch", exist_ok=True, mode=0o700)
+    return json.dumps({
+        "directories": directories,
+        "index_path": "/var/db/system/truesearch/index",
+        "log_level": "info",
+        "max_file_size": 104857600,
+        "worker_count": 0,
+        "batch_size": 1000,
+        "archive": {
+            "max_depth": 2,
+            "max_entries": 1000,
+            "max_archive_size": 524288000,
+            "index_contents": True,
+            "extract_text": False,
+            "supported_formats": ["zip", "tar", "gz", "bz2", "xz", "rar", "7z"]
+        },
+        "index_settings": {
+            "max_index_size": 10737418240,
+            "max_document_count": 1000000
+        },
+        "maintenance": {
+            "enabled": True,
+            "schedule": "0 2 * * *"
+        },
+        "reindex": {
+            "enabled": False,
+            "schedule": "weekly",
+            "interval_hours": 168,
+            "start_time": "02:00",
+            "max_duration_minutes": 240,
+            "clean_index": False
+        },
+        "service": {
+            "socket_path": "/run/truesearch/truesearch.sock",
+            "capability_mode": "read_only",
+            "required_capabilities": ["CAP_DAC_READ_SEARCH"],
+            "optional_capabilities": []
+        },
+        "security": {
+            "processing_timeout_seconds": 300,
+            "max_compression_ratio": 100.0,
+            "max_decompressed_size": 10737418240,
+            "max_memory_per_file": 104857600,
+            "enable_panic_recovery": True,
+            "validate_archive_paths": True,
+            "max_text_file_size": 10485760,
+            "restart_workers_on_panic": True,
+            "validate_capabilities": True,
+            "fail_without_capabilities": True,
+            "monitor_capability_changes": True,
+            "max_file_descriptors": 65536,
+            "enable_seccomp": False
+        },
+        "audit": {
+            "enabled": False,
+            "log_path": "/var/log/truesearch/audit.log",
+            "max_log_size": 104857600,
+            "sensitive_paths": ["/etc", "/root", "/home/*/.ssh"],
```

Comment:
What are sensitive_paths? Seems like there are actually a lot more than these on the server. Shouldn't it whitelist rather than blacklist?

Reply:
Ok, this is a placeholder for when we intended to later on hook up to your auditing framework @anodos325 . The examples here are a bit misleading, but the idea was to give the user the capability to flag specific paths as triggering auditing, vs others where maybe we don't care. Like if you have /mnt/tank/share1 and /mnt/tank/share-private defined, and want to explicitly enable auditing actions for queries into /mnt/tank/share-private

As it is, its a NOOP right now, so you can remove that if you like @themylogin - In the meantime open to suggestions if we want to just remove entirely, or punt until later when auditing becomes a thing.

Reply:
Yes, I will do same here. It'll just be enabled going forward, no config knobs.